pkg/build/webhook/webhook.go

```diff
@@ -47,53 +52,86 @@ func GitRefMatches(eventRef, configRef string, buildSource *buildapi.BuildSource
 	return configRef == eventRef
 }
 
-// FindTriggerPolicy retrieves the BuildTrigger of a given type from a build
-// configuration
-func FindTriggerPolicy(triggerType buildapi.BuildTriggerType, config *buildapi.BuildConfig) (buildTriggers []buildapi.BuildTriggerPolicy, err error) {
-	buildTriggers = buildapi.FindTriggerPolicy(triggerType, config)
-	if len(buildTriggers) == 0 {
-		err = ErrHookNotEnabled
-	}
-	return buildTriggers, err
+// NewWarning returns an StatusError object with a http.StatusOK (200) code.
+func NewWarning(message string) *kerrors.StatusError {
+	return &kerrors.StatusError{ErrStatus: metav1.Status{
+		Status:  metav1.StatusSuccess,
+		Code:    http.StatusOK,
+		Message: message,
+	}}
 }
 
-// ValidateWebHookSecret validates the provided secret against all currently
-// defined webhook secrets and if it is valid, returns its information.
-func ValidateWebHookSecret(webHookTriggers []buildapi.BuildTriggerPolicy, secret string) (*buildapi.WebHookTrigger, error) {
-	for _, trigger := range webHookTriggers {
-		switch trigger.Type {
-		case buildapi.GenericWebHookBuildTriggerType:
-			if !hmac.Equal([]byte(trigger.GenericWebHook.Secret), []byte(secret)) {
-				continue
-			}
-			return trigger.GenericWebHook, nil
-		case buildapi.GitHubWebHookBuildTriggerType:
-			if !hmac.Equal([]byte(trigger.GitHubWebHook.Secret), []byte(secret)) {
-				continue
+// CheckSecret tests the user provided secret against the secrets for the webhook triggers, if a match is found
+// then the corresponding webhook trigger is returned.
+func CheckSecret(namespace, userSecret string, triggers []*buildapi.WebHookTrigger, secretsClient kcoreclient.SecretsGetter) (*buildapi.WebHookTrigger, error) {
+	for i := range triggers {
+		secretRef := triggers[i].SecretRef
+		secret := triggers[i].Secret
+		if len(secret) > 0 {
+			if hmac.Equal([]byte(secret), []byte(userSecret)) {
+				return triggers[i], nil
 			}
-			return trigger.GitHubWebHook, nil
-
-		case buildapi.GitLabWebHookBuildTriggerType:
-			if !hmac.Equal([]byte(trigger.GitLabWebHook.Secret), []byte(secret)) {
-				continue
+		}
+		if secretRef != nil {
```

Comment:
didn't the API comment say this took precedence?

Reply:
it does, but that comment should be removed.  There's no concept of precedence, if you have both, we will accept a match on either one.

Reply:
> it does, but that comment should be removed. There's no concept of precedence, if you have both, we will accept a match on either one.

Ok. Update the comment.